compendium/DeclarativeServices/src/ComponentContextImpl.hpp

```diff
@@ -222,7 +247,11 @@ namespace cppmicroservices
 
             std::weak_ptr<ComponentConfiguration> configManager;
             cppmicroservices::Bundle usingBundle;
-            mutable Guarded<std::unordered_map<std::string, std::vector<cppmicroservices::InterfaceMapConstPtr>>>
+            // map of reference name to std::vector of pairs of ServiceReferenceBase to std::shared_ptr<void> of bound
+            // service in order that they were added to the boundServiceCache
+            mutable Guarded<std::unordered_map<
+                std::string,
+                std::vector<std::pair<cppmicroservices::ServiceReferenceBase, InterfaceMapConstPtr>>>>
```

Comment:
Any reason not to make this an `unordered_map<ServiceReferenceBase, InterfaceMapConstPtr>` ?
```suggestion
                std::unordered_map<cppmicroservices::ServiceReferenceBase, InterfaceMapConstPtr>>>
```

Reply:
yep. the ordering is used to determine which has the lower service id... locateService relies on that ordering... 

but that is a second conversation to have later on. In theory `LocateService` (the old one) returns the highest ranked service. This is not true at all... I think this used to be the case when dynamic binding didn't exist and the cache was the cache and never changed and the initial bound services which are retrieved from teh context are returned in order so by virtue of that, the first one is the highest ranked, but with dynamic rebinding and changes to the cache this is just not true anymore. However, that is probably separate from this pr and can be addressed separately.

Reply:
Ah yes, I see - that definitely sounds separate PR-worthy! Thanks :)